rasa/core/agent.py

```diff
@@ -400,6 +400,34 @@ def is_ready(self):
             and self.interpreter is not None
         )
 
+    async def parse_message_using_nlu_interpreter(self, message_data: Text):
```

Comment:
please add the return type. I believe it should be sth like that

```suggestion
    async def parse_message_using_nlu_interpreter(self, message_data: Text) -> Dict[Text, Any]:
```